### ISSUE TITLE
The problem matcher doesn't find all the errors

### ISSUE THREAD

**Issue by bbalp** [Microsoft/vscode#25257]
The problem matcher doesn't find all the errors. This is weird ?!

- VSCode Version: 1.11.2
- OS Version: Debian Jessie

Minimal steps to reproduce:

errors (it's a file to test the problem matcher)
```
../path/to/file.cpp:295:6: error: no matching function
../file.hpp:68:72: error: no match for call to
../path/to/file.cpp:301:50: error: has incomplete type
```
tasks.json
```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558
    // for the documentation about the tasks.json format
    "version": "0.1.0",
    "command": "cat",
    "isShellCommand": true,
    "tasks": [
        {
            "taskName": "errors",
            "isBuildCommand": true,
            "showOutput": "always",
            "problemMatcher": {
                "owner": "cpp",
                "pattern": {
                    "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
                    "file": 1,
                    "line": 2,
                    "column": 3,
                    "severity": 4,
                    "message": 5
                }
            }
        }
    ]
}
```
2 errors are found after running the task, not 3, the first error is missing.

But this version works fine:
```
../path/file.cpp:295:6: error: no matching function
../file.hpp:68:72: error: no match for call to
../path/to/file.cpp:301:50: error: has incomplete type
```

**Comment by jeremieca** [comment]
Same problem for me.  

When several errors have the same file path, only the last error is displayed, others are ignored.

**Comment by dbaeumer** [comment]
The problem is how we publish the problems. We publish problems for ../path/to/file.cpp when we encounter a new file to be processed. Since ../path/to/file.cpp adds additional problems later on they replace previous problems.

This was done to allow problems to be published as early as possible. Otherwise publishing problems has to wait until the task fully finished or we need to define a trigger to do so. In the C++ case such a trigger could be an error in another cpp file to honor error reporting from include files.

**Comment by dbaeumer** [comment]
To clarify
```
../path/to/file.cpp:295:6: error: no matching function
../path/to/file.cpp:301:50: error: has incomplete type
```

will produce both errors for file.cpp.

**Comment by bbalp** [comment]
Thank you for the explanation. And I confirm the case you posted, I tested it before posting the issue, it's the reason I gave up to guess what was going wrong, preferring to qualify the bug as a weird one. :)

**Comment by dbaeumer** [comment]
I need to move this. Priority for May was task v2.0 API and I didn't find time to look into this.

**Comment by dbaeumer** [comment]
Focus was again task 2.0.0

**Comment by dbaeumer** [comment]
Got fixed for August release (1.15).
